src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -198,10 +198,56 @@ private void SetupCommonOptionsForCommands(List<Command> commands)
                 return (int)ExitCode.ErrorInvalidARMTemplate;
             }
 
-            var analysisResult = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            ExitCode exitCode;
+            if (parametersFilePath != null)
+            {
+                exitCode = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            }
+            else
+            {
+                // Check if a parameters.json file is present according to naming standards here https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files#file-name, and if so use it as the parametersFile input
+                var parametersFiles = FindParameterFileForTemplate(templateFilePath);
+                int numOfFilesAnalyzed = 0;
+                bool issueReported = false;
+                var filesFailed = new List<FileInfo>();
+
+                if (parametersFiles.Count() > 0)
+                {
+                    foreach (FileInfo parametersFile in parametersFiles)
```

Comment:
AnalyzeDirectory calls AnalyzeTemplate, and the parameter search is done in both, so if the tool is called with analyze-directory then for each parameter-template combination we would look again at all the combinations

Reply:
Not sure if this is correct, but maybe I am missing something.
From looking at the code we have these relevant methods here:
- `AnalyzeDirectoryCommandHandler`
- `AnalyzeTemplateCommandHandler`
- `AnalyzeTemplate`
- `FindParameterFileForTemplate`

`AnalyzeDirectoryCommandHandler` and `AnalyzeTemplateCommandHandler` both call `AnalyzeTemplate`.
`AnalyzeDirectoryCommandHandler` and `AnalyzeTemplateCommandHandler` both call `FindParameterFileForTemplate`.

`FindParameterFileForTemplate` is not called from `AnalyzeTemplate` as far as I can see. 

If the tool is called with the `analyze-directory` command, the `AnalyzeDirectoryCommandHandler` will search for available parameters files using `FindParameterFileForTemplate`, and then for each combination of template and parameter file call `AnalyzeTemplate`. I don't think we have a situation here where the parameters files are searched for twice. 
Happy to hear of I'm wrong on this though 👍🏻